cupy/cuda/function.pyx

```diff
@@ -84,18 +84,13 @@ cdef set _pointer_numpy_types = {numpy.dtype(i).type
                                  for i in '?bhilqBHILQefdFD'}
 
 
+# A shortcut of Arg.from_obj(x).get_pointer() for the performance.
 cdef inline CPointer _pointer(x):
     cdef Py_ssize_t itemsize
     if x is None:
         return CPointer()
     if isinstance(x, core.ndarray):
         return (<core.ndarray>x).get_pointer()
```

Comment:
Dumb question (still trying to figure out what happened with the new `Arg` 😅): Isn't there a `NdarrayArg`? Why not also removing these two lines?

Reply:
This function is called only from [here](https://github.com/cupy/cupy/pull/3253/files#diff-411c12702aaf92fad2422d7ed44adf64R140) now and `NdarrayArg` does not reach here.
All indexers and scalars are converted to `IndexerArg` and `ScalarArg`, so these lines can be removed.

Reply:
I see, thanks @asi1024. I am still fuzzy about when exactly `NdarrayArg` is needed, though...A few pointers from you are greatly appreciated!